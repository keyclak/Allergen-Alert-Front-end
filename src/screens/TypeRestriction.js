import React, { useContext, useEffect, useState } from 'react';
import { TouchableOpacity, FlatList, View, Text, Pressable, StyleSheet} from 'react-native';
import LoadingButton from '../components/LoadingButton';
import TextLoadingButton from '../components/TextLoadingButton';
import { Colors, StyleConstants, Styles } from '../style';
import { useAddModification } from '../hooks/api';
import { ListItem, Icon } from 'react-native-elements'
import FormTextInput from '../components/FormTextInput';



export default function TypeRestriction({navigation}) {

    const [ingredient, setIngredient] = useState();
    const [type, setType] = useState();
    const addModification = useAddModification(ingredient, type);

    function onAddType0() {
        setType(0);
        addModification.execute()
            .then(() => navigation.pop())
            .catch(e => {});
    }

    function onAddType1() {
        setType(1);
        addModification.execute()
            .then(() => navigation.pop())
            .catch(e => {});
    }

    return (

        <View style={[Styles.container, {justifyContent: 'flex-start', paddingTop:40}]}>
            <View style={{width: StyleConstants.FormWidth}}>
                <FormTextInput placeholder="Ingredient" onChangeText={setIngredient}/>
                <View style={{flexDirection: "row", justifyContent: "space-between",paddingTop: 20}}>
                    <TextLoadingButton style={[Styles.button, {width: "47%"}]} text="Add to Diet Type 1" isLoading={addModification.loading} onPress={onAddType0} />
                    <TextLoadingButton style={[Styles.button, {width: "47%"}]} text="Add to Diet Type 2" isLoading={addModification.loading} onPress={onAddType1} />
                </View>
            </View>
        </View>
    );
}