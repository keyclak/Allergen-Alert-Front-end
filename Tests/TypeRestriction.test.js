import renderer from 'react-test-renderer';
import React from 'react';
import MockDate from 'mockdate'
import TypeRestriction from '../src/screens/TypeRestriction.js';

const FRAME_TIME = 10

const advanceOneFrame = () => {
  const now = Date.now();
  MockDate.set(new Date(now + FRAME_TIME));
  jest.advanceTimersByTime(FRAME_TIME);
}

const timeTravel = (msToAdvance = FRAME_TIME) => {
  const numberOfFramesToRun = msToAdvance / FRAME_TIME;
  let framesElapsed = 0;

  while (framesElapsed < numberOfFramesToRun) {
    advanceOneFrame();
    framesElapsed++;
  }
}
global.requestAnimationFrame = cb => {
  setTimeout(cb, FRAME_TIME);
}

export const setupTimeTravel = () => {
    MockDate.set(0);
    jest.useFakeTimers();
}

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.mock('@expo/vector-icons', () => ({
    MaterialIcons: 'Icon'
}));

beforeEach(setupTimeTravel);

const navigation = {
    addListener: jest.fn()
}

describe("TypeRestrictionPageTests", () => {
    test("Renders correctly", async () => {
        const tree = renderer.create(
            <TypeRestriction navigation={navigation}/>
        ).toJSON(); 

        timeTravel(500);

        expect(tree).toMatchSnapshot();
    })
})