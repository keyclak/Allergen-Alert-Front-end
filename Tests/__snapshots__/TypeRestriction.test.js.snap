// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeRestrictionPageTests Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "justifyContent": "flex-start",
        "paddingTop": 40,
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Ingredient"
      placeholderTextColor="#868e96"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#e9ecef",
            "borderColor": "#868e96",
            "borderRadius": 15,
            "borderWidth": 0,
            "color": "#495057",
            "fontSize": 16,
            "height": 50,
            "marginBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
          },
          undefined,
          false,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#2AB7CA",
              "borderRadius": 15,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 15,
            },
            false,
            Array [
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#2AB7CA",
                "borderRadius": 15,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 15,
              },
              Object {
                "width": "47%",
              },
            ],
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 200,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Add Exception
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#2AB7CA",
              "borderRadius": 15,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 15,
            },
            false,
            Array [
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#2AB7CA",
                "borderRadius": 15,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 15,
              },
              Object {
                "width": "47%",
              },
            ],
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 200,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Add Restriction
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          undefined,
          Object {
            "alignSelf": "center",
          },
        ]
      }
    />
  </View>
</View>
`;
